src/Hl7.Fhir.ElementModel/ElementNode.cs

```diff
@@ -0,0 +1,220 @@
+﻿/* 
+ * Copyright (c) 2018, Firely (info@fire.ly) and contributors
+ * See the file CONTRIBUTORS for details.
+ * 
+ * This file is licensed under the BSD 3-Clause license
+ * available at https://raw.githubusercontent.com/ewoutkramer/fhir-net-api/master/LICENSE
+ */
+
+using Hl7.Fhir.Specification;
+using Hl7.Fhir.Utility;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+
+namespace Hl7.Fhir.ElementModel
+{
+    public class DomNode<T> : IAnnotatable where T:DomNode<T>
+    {
+        public string Name { get; set; }
+
+        protected List<T> ChildrenInternal = new List<T>();
+
+        internal IEnumerable<T> ChildrenByName(string name = null) =>
```

Comment:
This name is a bit confusing: When you want all children, then you leave name = null. Why not Children()?

Reply:
It's an internal method that will be called by the actual public Children() method that is part of subclasses of DomNode implementing ITypedElement.  Renaming this to Children() would make the actual ITypedElement.Children() shadow this function.

I did however find a mistake in doing this (SourceNode's Children() should just call base.ChildrenByName()), which I will need to fix.

Reply:
I renamed the method to ChildrenInternal(), and renamed the list to ChildList.